app/Http/Controllers/API/v1/TransportController.php

```diff
@@ -370,25 +371,40 @@ public function create(Request $request): JsonResponse {
                                             'destination' => ['required', 'numeric'],
                                             'departure'   => ['required', 'date'],
                                             'arrival'     => ['required', 'date'],
-                                            'force'       => ['nullable', 'boolean']
+                                            'force'       => ['nullable', 'boolean'],
+                                            'with'        => ['nullable', 'array', 'max:10'],
                                         ]);
+        if (isset($validated['with'])) {
+            $withUsers = User::whereIn('id', $validated['with'])->get();
+            foreach ($withUsers as $user) {
+                if (!Auth::user()?->can('checkin', $user)) {
+                    return $this->sendError('You are not allowed to checkin for the given user.', 403);
```

Comment:
imho the error response should contain an array or a comma separated list of all users that the requesting user is not allowed to check in